src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -281,7 +281,14 @@ private void registerCheck(int tokenId, AbstractCheck check) throws CheckstyleEx
      */
     private void registerCheck(String token, AbstractCheck check) throws CheckstyleException {
         if (check.isCommentNodesRequired()) {
-            tokenToCommentChecks.put(token, check);
+            Set<AbstractCheck> checks = tokenToCommentChecks.get(token);
+
+            if (checks == null) {
+                checks = new HashSet<>();
+                tokenToCommentChecks.put(token, checks);
+            }
+
+            checks.add(check);
```

Comment:
I believe that these lines can be shortened to:

```java
tokenToCommentChecks.computeIfAbsent(token, k -> new HashMap<>()).put(check);
```

Reply:
This fixed idea violation in other area. I'm not sure why it didn't complain about this line too.